tests/jvm/src/test/scala-2.12/scala/meta/tests/semanticdb/OverridesSuite.scala

```diff
@@ -0,0 +1,54 @@
+package scala.meta.tests
+package semanticdb
+
+import scala.meta.internal.semanticdb.OverrideMode
+import scala.meta.internal.semanticdb.SemanticdbMode
+
+class OverridesSuite extends DatabaseSuite(SemanticdbMode.Slim, overrides = OverrideMode.Def) {
+  overrides(
+    """
+       |trait A { def foo: Int }
+       |trait B { def foo: Int }
+       |class C extends A with B { def foo: Int = 1 }
+       |class D extends C { def foo: Int = 2 }
+       """.stripMargin,
+    """
+       |_empty_.D#foo()I.{
+       |  _empty_.C#foo()I.
+       |  _empty_.B#foo()I.
+       |  _empty_.A#foo()I.
+       |}
+       |_empty_.C#foo()I.{
+       |  _empty_.B#foo()I.
+       |  _empty_.A#foo()I.
+       |}""".stripMargin
+  )
+  overrides(
+     """
+       |object a {
+       |  val foo = new java.util.TimerTask {
+       |    def run(): Unit = ()
+       |  }
+       |}""".stripMargin,
+     """
+       |_empty_.a.foo.$anon#run()V.{
+       |  _root_.java.util.TimerTask#run()V.
+       |  _root_.java.lang.Runnable#run()V.
+       |}""".stripMargin
+  )
+  overrides(
+   """
+     |trait Foo {
+     |  def foo: Int
+     |}
+     |class Foobar extends Foo {
+     |  def foo = 3
+     |}
+   """.stripMargin,
+   """
+     |_empty_.Foobar#foo()I.{
+     |   _empty_.Foo#foo()I.
+     |}
+   """.stripMargin
+  )
+}
```

Comment:
Please, update the benchmark suite as follows:

```diff
diff --git a/bench/README.md b/bench/README.md
index a3e3a52e3c..da53d75a78 100644
--- a/bench/README.md
+++ b/bench/README.md
@@ -25,12 +25,14 @@ ScalametaBaseline corresponds to the following compiler plugin options:
 -P:semanticdb:mode:slim
 -P:semanticdb:denotations:definitions
 -P:semanticdb:synthetics:none
+-P:semanticdb:overrides:none

 Other Scalameta configurations progressively enable more and more features:
   * ScalametaFullContents enables `-P:semanticdb:mode:fat`
   * ScalametaFullDenotations further enables `-P:semanticdb:denotations:all`
   * ScalametaFullSynthetics further enables `-P:semanticdb:synthetics:all`
+  * ScalametaFullOverrides further enables `-P:semanticdb:overrides:all`

 ## Results

@@ -43,4 +45,5 @@ QuickScalametaBaseline         sample  15  2020.256 ± 39.586  ms/op  (25% overh
 QuickScalametaFullContents     sample  15  2081.549 ± 38.977  ms/op  (30% overhead)
 QuickScalametaFullDenotations  sample  15  2236.738 ± 28.241  ms/op  (39% overhead)
 QuickScalametaFullSynthetics   sample  15  2272.054 ± 81.361  ms/op  (42% overhead)
+QuickScalametaFullOverrides    sample  ??  ????.??? ± ??.???  ms/op  (??% overhead)

diff --git a/bench/suite/src/main/scala/org/scalameta/bench/Scalameta.scala b/bench/suite/src/main/scala/org/scalameta/bench/Scalameta.scala
index 22422def38..e5cd8089d9 100644
--- a/bench/suite/src/main/scala/org/scalameta/bench/Scalameta.scala
+++ b/bench/suite/src/main/scala/org/scalameta/bench/Scalameta.scala
@@ -50,6 +50,26 @@ trait Scalameta {
 @Warmup(iterations = 5, time = 10, timeUnit = TimeUnit.SECONDS)
 @Measurement(iterations = 5, time = 10, timeUnit = TimeUnit.SECONDS)
 @Fork(value = 1, jvmArgs = Array("-Xms2G", "-Xmx2G"))
+class QuickScalametaFullOverrides extends Scalameta {
+  @Benchmark
+  def run(bs: BenchmarkState): Unit = {
+    runImpl(bs)
+  }
+  override def mkSettings(bs: BenchmarkState): Settings = {
+    val settings = super.mkSettings(bs)
+    settings.pluginOptions.value ::= "semanticdb:overrides:all"
+    settings.pluginOptions.value ::= "semanticdb:synthetics:all"
+    settings.pluginOptions.value ::= "semanticdb:denotations:all"
+    settings.pluginOptions.value ::= "semanticdb:mode:fat"
+    settings
+  }
+}
+
+@BenchmarkMode(Array(SampleTime))
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Warmup(iterations = 5, time = 10, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 5, time = 10, timeUnit = TimeUnit.SECONDS)
+@Fork(value = 1, jvmArgs = Array("-Xms2G", "-Xmx2G"))
 class QuickScalametaFullSynthetics extends Scalameta {
   @Benchmark
   def run(bs: BenchmarkState): Unit = {
@@ -57,6 +77,7 @@ class QuickScalametaFullSynthetics extends Scalameta {
   }
   override def mkSettings(bs: BenchmarkState): Settings = {
     val settings = super.mkSettings(bs)
+    settings.pluginOptions.value ::= "semanticdb:overrides:none"
     settings.pluginOptions.value ::= "semanticdb:synthetics:all"
     settings.pluginOptions.value ::= "semanticdb:denotations:all"
     settings.pluginOptions.value ::= "semanticdb:mode:fat"
@@ -76,6 +97,7 @@ class QuickScalametaFullDenotations extends Scalameta {
   }
   override def mkSettings(bs: BenchmarkState): Settings = {
     val settings = super.mkSettings(bs)
+    settings.pluginOptions.value ::= "semanticdb:overrides:none"
     settings.pluginOptions.value ::= "semanticdb:synthetics:none"
     settings.pluginOptions.value ::= "semanticdb:denotations:all"
     settings.pluginOptions.value ::= "semanticdb:mode:fat"
@@ -95,6 +117,7 @@ class QuickScalametaFullContents extends Scalameta {
   }
   override def mkSettings(bs: BenchmarkState): Settings = {
     val settings = super.mkSettings(bs)
+    settings.pluginOptions.value ::= "semanticdb:overrides:none"
     settings.pluginOptions.value ::= "semanticdb:synthetics:none"
     settings.pluginOptions.value ::= "semanticdb:denotations:definitions"
     settings.pluginOptions.value ::= "semanticdb:mode:fat"
@@ -114,6 +137,7 @@ class QuickScalametaBaseline extends Scalameta {
   }
   override def mkSettings(bs: BenchmarkState): Settings = {
     val settings = super.mkSettings(bs)
+    settings.pluginOptions.value ::= "semanticdb:overrides:none"
     settings.pluginOptions.value ::= "semanticdb:synthetics:none"
     settings.pluginOptions.value ::= "semanticdb:denotations:definitions"
     settings.pluginOptions.value ::= "semanticdb:mode:slim"
```

Then, run `bin/bench All` to update the benchmark results.

Reply:
@xeno-by what is the JVM you are using?

Reply:
When I run the benchmark I get really strange output:

```
[info] # JMH version: 1.19
[info] # VM version: JDK 1.8.0_152, VM 25.152-b16
[info] # VM invoker: /nix/store/27dcyklrbxci01vmvywwjanr60d418fy-openjdk-8u152b16-jre/lib/openjdk/jre/bin/java
[info] # VM options: -Xms2G -Xmx2G
[info] # Warmup: 5 iterations, 10 s each
[info] # Measurement: 5 iterations, 10 s each
[info] # Timeout: 10 min per iteration
[info] # Threads: 1 thread, will synchronize iterations
[info] # Benchmark mode: Sampling time
[info] # Benchmark: org.scalameta.bench.QuickScalacRangepos.run
[info] 
[info] # Run progress: 50.00% complete, ETA 00:01:46
[info] # Fork: 1 of 1
[info] # Warmup Iteration   1: 3918.878 ±(99.9%) 44650.835 ms/op
[info] # Warmup Iteration   2: 1539.010 ±(99.9%) 476.396 ms/op
[info] # Warmup Iteration   3: 1301.283 ±(99.9%) 129.220 ms/op
[info] # Warmup Iteration   4: 1166.949 ±(99.9%) 41.814 ms/op
[info] # Warmup Iteration   5: 1255.145 ±(99.9%) 160.903 ms/op
```

Do you know why the ms/op drops like that?

Reply:
I'm using the latest HotSpot 1.8 on macOS. Never seen anything like this...